contracts/mocks/MockValidator.sol

```diff
@@ -2,36 +2,50 @@
 pragma solidity ^0.8.27;
 
 import { IModule } from "../interfaces/modules/IModule.sol";
-import { IValidator } from "../interfaces/modules/IValidator.sol";
 import { IModuleManager } from "../interfaces/base/IModuleManager.sol";
 import { VALIDATION_SUCCESS, VALIDATION_FAILED, MODULE_TYPE_VALIDATOR, ERC1271_MAGICVALUE, ERC1271_INVALID } from "../types/Constants.sol";
 import { PackedUserOperation } from "account-abstraction/contracts/interfaces/PackedUserOperation.sol";
 import { ECDSA } from "solady/src/utils/ECDSA.sol";
 import { SignatureCheckerLib } from "solady/src/utils/SignatureCheckerLib.sol";
 import { MessageHashUtils } from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import { ERC7739Validator } from "../base/ERC7739Validator.sol";
+import { IERC1271Unsafe } from "../interfaces/modules/IERC1271Unsafe.sol";
 
-contract MockValidator is IValidator {
+contract MockValidator is ERC7739Validator, IERC1271Unsafe {
     mapping(address => address) public smartAccountOwners;
 
     function validateUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash) external view returns (uint256 validation) {
-        return
-            ECDSA.recover(MessageHashUtils.toEthSignedMessageHash(userOpHash), userOp.signature) == smartAccountOwners[msg.sender]
-                ? VALIDATION_SUCCESS
-                : VALIDATION_FAILED;
+        address owner = smartAccountOwners[msg.sender];
+        return _validateSignatureForOwner(owner, userOpHash, userOp.signature) ? VALIDATION_SUCCESS : VALIDATION_FAILED;
```

Comment:
owner can be inlined into the return to save some gas

Reply:
inlined for K1
for Mock left as is for verbosity as it isn't going to prod